tests/e2e/vmc_csi_deployments.go

```diff
@@ -0,0 +1,215 @@
+		gomega.Expect((dep.Status.Replicas) == replica).To(gomega.BeTrue(), "Number of containers in the deployment should match with number of replicas")
+
+		ginkgo.By("Delete Deployment")
+		err = client.AppsV1().Deployments(namespace).Delete(ctx, deploymentName, metav1.DeleteOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	})
+
+	/*
+		Test performs following operations
+
+		Steps
+		1. Create Storage Class and PVC
+		2. Deploy nginx pods with volume
+		3. Update the nginx deployment pods, update the nginx image from k8s.gcr.io/nginx-slim:0.8 to k8s.gcr.io/nginx-slim:0.9
+		4. Wait for some time and verify the update is successful
+		5. Scale dowm the deployment to 0 replicas
+		6. Scale up the deployment to 1 replicas and verify all the pods should be up and running
+		7. Delete the deployments and Verify deployments are deleted successfully
+		8. Delete the PVC
+	*/
+	ginkgo.It("Deployments with volume provisioning", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		var lables = make(map[string]string)
+		lables["app"] = "nginx"
+		deploymentName = "testdeployment"
+		replica = 1
+		maxReplica = 0
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+
+		ginkgo.By("Creating StorageClass for Deployment")
+		sc, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating PVC")
+		pvclaims = append(pvclaims, pvclaim)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, err = fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvclaim = nil
+		}()
+
+		ginkgo.By("Creating Deployments")
+		deploymentSpec := NewDeploymentwithVolume(deploymentName, replica, lables, "nginx", "k8s.gcr.io/nginx-slim:0.8", pvclaims, "RollingUpdate")
+		createDeployments(namespace, deploymentSpec, client)
+
+		ginkgo.By("Update deployment image")
+		updateDeploymentImage(namespace, "k8s.gcr.io/nginx-slim:0.9", deploymentName)
+
+		ginkgo.By("Scale down replicas")
+		updateDeploymentReplica(client, maxReplica, deploymentName, namespace)
+		dep, err := client.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("replica count after scale down of deployments %v", dep.Status.Replicas)
+		gomega.Expect((dep.Status.Replicas) == maxReplica).To(gomega.BeTrue(), "Number of containers in the deployment should match with number of replicas")
```

Comment:
We should also verify if the required number of replicas are running ready as well

Reply:
as discussed I removed this test cases